### ISSUE TITLE
even prettier → encoding error

### ISSUE THREAD

**Issue by hoodie** [phsym/prettytable-rs#35]
**I have rewritten the original issue**

Hey there,
I would love to draw line separators with these symbols:

```rust
.separators( &[format::LinePosition::Top],    format::LineSeparator::new('─', '┬', '┌', '┐'))
.separators( &[format::LinePosition::Intern], format::LineSeparator::new('─', '┼', '├', '┤'))
.separators( &[format::LinePosition::Bottom], format::LineSeparator::new('─', '┴', '└', '┘'))
```
however, there seems to be some sort of encoding problem, because they are not printed this way on the terminal

## Expectation:
```
┌─────────────┬────────────┐
│ Title 1     | Title 2    │ 
├─────────────┼────────────┤
│ Value 1     | Value 2    │ 
├─────────────┼────────────┤
│ Value three | Value four │ 
└─────────────┴────────────┘
```
## Result:
```
,
 Title 1      Title 2    
<$
 Value 1      Value 2    
<$
 Value three  Value four 
4▒
```

Do you have an idea what goes wrong here?
Thanks

**Comment by hoodie** [comment]
The problem occurs because in https://github.com/phsym/prettytable-rs/blob/master/src/format.rs#L56 the chars which are infact [four bytes](https://doc.rust-lang.org/stable/std/primitive.char.html#representation) are force into a `u8`. this could be fixed by taking `&str` instead of `char` or and passing something like `	try!(out.write_all(self.ljunc.as_bytes()));`, which however would change the API.

**Comment by hoodie** [comment]
ok, I've played around with few alternatives. Unfortunately `char` does not have an `as_bytes()` however I don't get why.
The [here](https://github.com/rust-lang/rust/issues/27784) `unicode` feature should enable [this version](https://github.com/hoodie/prettytable-rs/tree/bug/encode_utf8) which might be the cheapest version, I've alternatively you could [temporarily convert to `String`](https://github.com/hoodie/prettytable-rs/tree/bug/encode_tostring) until the `unicode` feature gets here. I would propose a complete switch to `str`s, but I get that this would entail alot of [breaking changes](https://github.com/hoodie/prettytable-rs/tree/bug/linesep_strings) :disappointed:

So yeah, feel free to pull any of these three branches :smile:

**Comment by nabijaczleweli** [comment]
> Unfortunately char does not have an as_bytes() however I don't get why.

Because [`char` is `u32`](https://is.gd/cWCPay)

**Comment by hoodie** [comment]
@nabijaczleweli that's true, but couldn't a `u32` still be split into four `u8`? But perhaps there are still some loopholes to that idea

**Comment by nabijaczleweli** [comment]
See? Now you implemented `char::as_bytes()`, problem solved.

**Comment by hoodie** [comment]
I guess I want some sort of safe abstraction from the stdlib, otherwise you can't say *"This library uses 100% safe code"* :P 
And I wouldn't have written anything unsafe in PR either, even though I suppose this is perhaps the most efficient solution to the problem.

**Comment by phsym** [comment]
Hi guys ! I just tried the crate [**encode_unicode**](https://crates.io/crates/encode_unicode). Looks like it could be use to easily fix this issue

**Comment by nabijaczleweli** [comment]
@hoodie Just implement an equivalent yourself v0v

**Comment by nabijaczleweli** [comment]
@hoodie Just implement an equivalent yourself like this: https://is.gd/sIb5CA

**Comment by phsym** [comment]
@hoodie In fact, this is 99% safe code. see [https://github.com/phsym/prettytable-rs/blob/master/src/lib.rs#L427] lib.rs :wink:

**Comment by nabijaczleweli** [comment]
```rust
&mut *((self as *const Table) as *mut Table)
```
kill me

**Comment by hoodie** [comment]
publish s'il vous plait

**Comment by phsym** [comment]
I'll do after lunch

**Comment by hoodie** [comment]
😜 bon appetit

**Comment by phsym** [comment]
merci

**Comment by phsym** [comment]
Et voilà ! Published in v0.6.4
